community/modules/scripts/pbspro-preinstall/main.tf

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  location = coalesce(var.location, var.region)
+
+  bucket_name = "pbspro-packages"
+
+  retention_policy = {
+    (local.bucket_name) = var.retention_policy
```

Comment:
I've not used this syntax in TF before, what is the purpose of the () around local.bucket_name?

Reply:
Because quotation marks are optional (and generally not preferred) for fields/keys in objects/maps, it is the only way to guide Terraform into realizing that you want the resolved value for that variable.

Reply:
https://developer.hashicorp.com/terraform/language/expressions/types#maps-objects